File: lib/src/main/java/com/chineseall/epubparser/lib/html/ContainerNode.kt
package com.chineseall.epubparser.lib.html

import com.chineseall.epubparser.lib.render.RenderContext
import com.chineseall.epubparser.lib.render.RenderSection

// 容器节点
class ContainerNode(
    nodeType: String?,
    type: String?,
    className: String?,
    style: String?,
    bookPlot: BookPlot?,
    var children: MutableList<BaseNode>? = null
) : BaseNode(nodeType, type, className, style, bookPlot) {

    override fun appendSection(renderContext: RenderContext, sections: MutableList<RenderSection>) {
        children?.let { nodes ->
            for (node in nodes) {
                node.appendSection(renderContext, sections)
            }
        }
    }
}

File: js_project/client/src/epubfile/html/ContainerNode.js
import BaseNode from './BaseNode'
// 容器类型节点 可以包含子节点
export default class ContainerNode extends BaseNode {
    // 子节点列表
    children
    constructor(props){
        super(props)
        this.nodeType = 'container'
        this.children = props && props.children
    }
}